--- a/CMakeLists.txt	2026-05-01 14:56:34
+++ b/CMakeLists.txt	2026-05-01 15:51:39
@@ -21,7 +21,7 @@
     -s USE_LIBPNG=1 \
     -s ALLOW_MEMORY_GROWTH=1 \
     -s NO_EXIT_RUNTIME=1 \
-    -s EXTRA_EXPORTED_RUNTIME_METHODS=['FS'] \
+    -s EXTRA_EXPORTED_RUNTIME_METHODS=['FS','UTF8ToString'] \
     --no-heap-copy")
 
 if (CMAKE_BUILD_TYPE MATCHES Debug)
--- a/src/v_trans.h	2026-05-01 14:26:57
+++ b/src/v_trans.h	2026-05-01 15:32:43
@@ -35,7 +35,7 @@
 
 #include "doomtype.h"
 
-enum
+typedef enum
 {
     CR_NONE,
     CR_DARK,
@@ -48,7 +48,6 @@
     CR_RED2GREEN,
     CRMAX
 } cr_t;
-
 extern byte *cr[CRMAX];
 extern char **crstr;
 
--- a/src/i_gif.c	2026-05-01 14:26:57
+++ b/src/i_gif.c	2026-05-01 15:51:28
@@ -31,7 +31,7 @@
     gif = NULL;
 
     EM_ASM({
-        var filename = Module.Pointer_stringify($0);
+        var filename = Module.UTF8ToString($0);
         var url = URL.createObjectURL(new Blob([Module.FS.readFile(filename)], {type: 'image/gif'}));
         document.dispatchEvent(new CustomEvent("I_CloseGIF", { detail: { url: url } }));
         Module.FS.unlink(filename);
--- a/src/i_system.c	2026-05-01 14:26:57
+++ b/src/i_system.c	2026-05-01 15:51:28
@@ -259,7 +259,7 @@
     va_end(argptr);
 
     EM_ASM_({
-        document.dispatchEvent(new CustomEvent("I_Error", { detail: { errorMsg: Module.Pointer_stringify($0) } }));
+        document.dispatchEvent(new CustomEvent("I_Error", { detail: { errorMsg: Module.UTF8ToString($0) } }));
     }, msgbuf);
 
     // Shutdown. Here might be other errors.
--- a/src/i_webmusic.c	2026-05-01 14:26:57
+++ b/src/i_webmusic.c	2026-05-01 15:51:28
@@ -1031,7 +1031,7 @@
                 delete window.doom_music_offset;
                 delete window.doom_music_start;
                 delete window.doom_music_looping;
-                var filename = Module.Pointer_stringify($0);
+                var filename = Module.UTF8ToString($0);
                 Module.FS.unlink("./" + filename);
             }catch(err){}
         }, current_filename);
@@ -1059,7 +1059,7 @@
         playing_substitute = true;
         current_filename = filename;
         EM_ASM_({
-            var filename = Module.Pointer_stringify($0);
+            var filename = Module.UTF8ToString($0);
             window.doom_music_filename = filename;
             fetch(filename).then(function(res){ return res.arrayBuffer(); }).then(function(audio){
                 var context = SDL2.audioContext;
--- a/src/v_video.c	2026-05-01 14:26:57
+++ b/src/v_video.c	2026-05-01 15:51:28
@@ -1009,7 +1009,7 @@
     fclose(handle);
 
     EM_ASM({
-        var filename = Module.Pointer_stringify($0);
+        var filename = Module.UTF8ToString($0);
         var url = URL.createObjectURL(new Blob([Module.FS.readFile(filename)], {type: 'image/png'}));
         document.dispatchEvent(new CustomEvent("V_ScreenShot", { detail: { url: url } }));
         Module.FS.unlink(filename);
--- a/src/doom/p_inter.c	2026-05-01 14:26:57
+++ b/src/doom/p_inter.c	2026-05-01 15:51:28
@@ -721,7 +721,7 @@
 
 	const char *s = HU_GetMapName();
 	EM_ASM_({
-        document.dispatchEvent(new CustomEvent("P_KillMobj", { detail: { mapname: Module.Pointer_stringify($0), source: Module.Pointer_stringify($1) } }));
+        document.dispatchEvent(new CustomEvent("P_KillMobj", { detail: { mapname: Module.UTF8ToString($0), source: Module.UTF8ToString($1) } }));
     }, s, t);
 
 	P_DropWeapon (target->player);
--- a/src/doom/g_game.c	2026-05-01 14:26:57
+++ b/src/doom/g_game.c	2026-05-01 15:51:28
@@ -684,7 +684,7 @@
     const char *s = HU_GetMapName();
 
     EM_ASM_({
-        document.dispatchEvent(new CustomEvent("G_DoLoadLevel", { detail: { mapname: Module.Pointer_stringify($0) } }));
+        document.dispatchEvent(new CustomEvent("G_DoLoadLevel", { detail: { mapname: Module.UTF8ToString($0) } }));
     }, s);
 } 
 
@@ -1550,7 +1550,7 @@
         var leveltime = $8;
         document.dispatchEvent(new CustomEvent("G_DoCompleted", {
             detail: {
-                mapname: Module.Pointer_stringify($0),
+                mapname: Module.UTF8ToString($0),
                 maxkills: maxkills,
                 maxitems: maxitems,
                 maxsecret: maxsecret,
@@ -1785,7 +1785,7 @@
 
     EM_ASM_({
         try{
-            var filename = Module.Pointer_stringify($0);
+            var filename = Module.UTF8ToString($0);
             var buffer = Module.FS.readFile(filename).buffer;
             document.dispatchEvent(new CustomEvent("G_SaveGame", { detail: { filename: filename, buffer: buffer } }));
         }catch(err){}
